include/dynamatic/Dialect/Handshake/HandshakeInterfaces.td

```diff
@@ -272,6 +268,25 @@ InterfaceMethod<[{
         return attr.getValue();
       }
       return FPUImpl::FLOPOCO;
+    }]>,
+
+  InterfaceMethod<[{
+      Sets the internal delay as a StringAttr named "internal_delay".
+    }], "void", "setInternalDelay", (ins "::llvm::StringRef":$value), [{
+      Operation *op = $_op.getOperation();
+      op->setAttr("internal_delay",
+                   StringAttr::get(op->getContext(), value));
+    }]>,
+    
+  // Return StringAttr instead of StringRef to ensure proper memory allocation.
+  // Using StringRef would result in the default value "0.0" becoming a dangling pointer.
+  InterfaceMethod<[{
+      Gets the internal delay as a StringAttr named "internal_delay". Defaults to "0.0" if not set.
+    }], "StringAttr", "getInternalDelay", (ins), [{
+      Operation *op = $_op.getOperation();
+      if (auto attr = op->getAttrOfType<StringAttr>("internal_delay"))
+        return attr;
+      return StringAttr::get(op->getContext(), "0.0");
```

Comment:
If `internal_delay` is stored in the format of `"0.0"`, do we need to convert it into `"0_0"` before passing to the RTL?

Reply:
I think this default value is wrong, the buffering pass sets the converted format:
https://github.com/EPFL-LAP/dynamatic/blob/d3799298d9136a1b5031fab6adc1ebde212ff035/lib/Transforms/BufferPlacement/HandshakePlaceBuffers.cpp#L257

So can you update this to `"0_0"`?